### ISSUE TITLE
When I installed with pip, I didn't end up with the bin folder - is that a problem?

### ISSUE THREAD

**Issue by keharr** [MrOlm/drep#186]
I installed drep with pip, and according to pip everything is absolutely 100% fine and it's downloaded successfully. But I don't end up with the bin folder, and thus also no dRep executable file, which I think is the reason why I can't actually tell it to do any commands but I can import drep to python. If that isn't the case, I'd love to know! I'm planning to use dRep to analyze some metagenomes but at this point I've been fighting it for three days in an unsuccessful attempt to figure out why it won't recognize the command as a command.

**Comment by MrOlm** [comment]
Hi @keharr 

How frustrating! Unfortunately I think this is a `pip` problem, not an inStrain problem. Pip should automatically make the python executable in the correct place on your system.

You could try and just download and run the `inStrain` python file from the `bin` folder in this repo (https://github.com/MrOlm/drep/blob/master/bin/dRep) but my guess is this won't work because of whatever `pip` issue you're facing with your system. That said, this has worked for me when I've hit `pip` issues in the past. 

I would recommend trying to install with conda, or even just using conda to create a new python virtual envrionment on your system and then installing with the new `pip` from within that python virtual environment.

Good luck and sorry for the troubles!
-Matt

**Comment by keharr** [comment]
I finally figured out what's going on I think! When pip installs, it does use the setup.py file to create a file that should be the executable, but the file it's creating has no extension. That file isn't Windows executable, regardless of what version of pip I install it with, or if I run pip in conda or without. I can't make it executable, either - no matter what I try my system rejects it. The lightbulb finally went on - am I correct in assuming that this was written for Linux, and not supported on Windows? If so, would you mind updating the documentation to include that fact? If it's in there, I couldn't find it.

**Comment by MrOlm** [comment]
dRep is a Python program, and Python can indeed be run on Windows. I don't know the details of how to get Python set up working on Windows, but once properly set up dRep can be installed and will run properly.

The only file dRep gives to `pip` is this one - https://github.com/MrOlm/drep/blob/master/setup.py . You can see it's just a short list of dependencies. The installation issues you're facing are related to the particular way Python is configured on your system. I would again encourage you to use conda as a way of addressing this problems - https://docs.conda.io/projects/conda/en/latest/user-guide/install/windows.html

Best,
Matt

**Comment by keharr** [comment]
Hi Matt,

I've used Python on Windows for a long time and I'm confident it's set up correctly! I tried to install dRep both within and outside of conda (my first place to start, but thanks for the suggestion!), and attempted to install it both with pip and with conda install. pip correctly built a file from setup.py, but the file starts with a shebang (#!) which is not acceptable syntax on Windows, and therefore can't send the correct command to run python. I tried a million variations of tweaking the file to fix this issue - on Windows, the correct form of setup would be for it to build a batch file (.bat) but dRep's builds without a file extension. Unfortunately, just correcting the file extension to .bat does not fix the problem. I also attempted to convert it to a .py file, which Windows does know what to do with, but then every single line in the program is not recognized correctly as instructions for python due to some quirk of how the file is created from the setup.py file! And I tried a bunch of other things which are neither here nor there, but all failed. It's possible that your setup file is no longer working with the current version of pip (which does say that a setup.py file is now not the recommended version of creating an executable).

Since you also suggested conda, when I tried setting up with a conda install on a clean environment (and I thought it might have been an environment, conda, or pip problem, so I tried uninstalling and reinstalling conda, making a second new environment, and updating all the relevant pieces), it did not install at all! I think I was likely running into the same issue as the person in November in issue #156, since a lot of their problems showed up in my troubleshooting process as well.

I appreciate your attempt to help with my problem, but I would strongly suggest that your program is not functional on Windows at the moment, and that Windows users should simply install it on a virtual machine and save their time.

Thank you,

keharr
